### PR TITLE
Add basic github actions for CI workflow 

### PR DESCRIPTION
**Pull request by umeshkumhar**
# Description

This PR add some github actions that run CI jobs to validate the PR code by running some  required actions
Golang code - run go lint 
Vue & Npm code - run npm build & run lint

This is beneficial for the new contributors in raising PRs and since cloudbuild jobs are private to only Google employees. 

## Primary Components Effected

Please mark one option with an "x".

- [ ] Shifter (Core CLI and Server Application)
- [ ] Shifter UI (The Web Interface)
- [x] Cross Component Configuration (.github workflows, templates, docs, .MD files)

## Type of change

Please mark one option with an "x".

- [ ] #RELEASE New Collection of Features (containing breaking and non-breaking changes)
- [ ] #FT New feature (non-breaking change which adds functionality, improves performance or code quality)
- [ ] #BUG Bug Fix (non-breaking change which fixes an issue)
- [ ] #PATCH Patch (non-breaking change which modifies only non-code object [Docs, Readme])

## Developer / Author Notes

Please feel free to provide any additional notes, content or context relating to this PR and subsequent contribution.

### CODE REVIEW

**Comment by garybowers** [comment]
We are not using giithub ci, could you please add this to cloudbuild instead.

File: shifter/shifter.go [hunk 1]
```diff
@@ -19,3 +19,5 @@ import "shifter/cmd"
 func main() {
 	cmd.Execute()
 }
+
```
Comment (garybowers) [inline shifter/shifter.go:22]: Why the added blank lines?

File: .github/workflows/golang-ci.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: golang-ci
```
Comment (garybowers) [inline .github/workflows/golang-ci.yml:1]: As mentioned earlier - we are not using github actions at all, this needs to be moved to the cloudbuild job

File: .github/workflows/npm-ci.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: npm-ci-workflow
```
Comment (garybowers) [inline .github/workflows/npm-ci.yml:1]: As mentioned earlier - we are not using github actions at all, this needs to be moved to the cloudbuild job

**Review by garybowers — CHANGES_REQUESTED** [review]
Remove github actions and move to cloudbuild as this is the project strategic choice of CI.

**Comment by umeshkumhar** [comment]
> Remove github actions and move to cloudbuild as this is the project strategic choice of CI.

Hey @garybowers 

We discussed this actually, this is basically CI jobs for external contributors, since cloud build jobs are restricted to core collaborators only. If external contributor raises PR then there is no way to validate that his/her PR is in good state to review.

So we were thinking to have github actions for basic PR readiness jobs such as linting issues, gosec issues..

And for other testing pipeline and release pipeline we can still be on cloud build yamls.

Let me know your thoughts on this.

File: shifter/shifter.go (hunk 1 above)
Comment (umeshkumhar) [reply shifter/shifter.go:22]: just testing out linting issues

**Review by garybowers — APPROVED** [review]
lgtm

**Pull request merged**